core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala

```diff
@@ -51,17 +49,16 @@ trait IntegrationTestHarness extends KafkaServerTestHarness {
     cfgs.map(KafkaConfig.fromProps)
   }
 
+  lazy val producerConfig = new Properties(TestUtils.producerSecurityConfigs(securityProtocol, trustStoreFile))
+  lazy val consumerConfig = new Properties(TestUtils.consumerSecurityConfigs(securityProtocol, trustStoreFile))
+
   @Before
   override def setUp() {
-    val producerSecurityProps = TestUtils.producerSecurityConfigs(securityProtocol, trustStoreFile)
-    val consumerSecurityProps = TestUtils.consumerSecurityConfigs(securityProtocol, trustStoreFile)
     super.setUp()
-    producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, classOf[org.apache.kafka.common.serialization.ByteArraySerializer])
-    producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, classOf[org.apache.kafka.common.serialization.ByteArraySerializer])
-    producerConfig.putAll(producerSecurityProps)
-    consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, classOf[org.apache.kafka.common.serialization.ByteArrayDeserializer])
-    consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, classOf[org.apache.kafka.common.serialization.ByteArrayDeserializer])
-    consumerConfig.putAll(consumerSecurityProps)
+    producerConfig.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer")
```

Comment:
Good idea.

I'm still working through untangling in the tests.  I'm finding there are a lot of uses of the Properties.put(String, Object) (which is from the HashMap that gets extended) as opposed to Properties.setProperty(String, String).  This is going to make my final diff a big large unfortunately.

(I didn't intend this for a WIP but the above misuse of Properties is making it much harder).